posthog/batch_exports/http.py

```diff
@@ -422,6 +423,34 @@ def backfill(self, request: request.Request, *args, **kwargs) -> response.Respon
         else:
             end_at = None
 
+        if start_at is not None or end_at is not None:
+            earliest_backfill_start_at = fetch_earliest_backfill_start_at(
+                team_id=self.team_id,
+                model=batch_export.model,
+                interval_time_delta=batch_export.interval_time_delta,
+                exclude_events=batch_export.destination.config.get("exclude_events", []),
+                include_events=batch_export.destination.config.get("include_events", []),
+            )
+            if earliest_backfill_start_at is None:
+                raise ValidationError("There is no data to backfill for this model.")
+
+            earliest_backfill_start_at = earliest_backfill_start_at.astimezone(self.team.timezone_info)
+
+            if end_at is not None and end_at < earliest_backfill_start_at:
+                raise ValidationError(
+                    "The provided backfill date range contains no data. The earliest possible backfill start date is "
+                    f"{earliest_backfill_start_at.strftime('%Y-%m-%d %H:%M:%S')}",
```

Comment:
nit: This should be adjusted based on interval. E.g. `'%Y-%m-%d'` for daily, `'%Y-%m-%d  %H'` (or nicer) for hourly, and so on.

Reply:
yeah agree, just left it out in the interest of time but good to keep in mind